Ad9545 add status debugfs

tachicialex opened the pull request:
clk: ad9545: Add debugfs status: Add debugfs files for the two PLL blocks, 4 input references, Auxiliary DPLL:

AUX_DPLL:
Lock status: Unlocked
Reference: Faulted

PLL0:
PLL status: Locked
Freerun Mode: Off
Holdover Mode: Off
PLL Profile: On
Profile Number: 0
Temperature: 49 C

Ref-A-Div:
Reference: Faulted: Too Fast Too Slow

clk: ad9545: Fix typo: fixed a typo in the ad9545_pll_clk struct

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -1402,6 +1420,53 @@ static int ad9545_outputs_setup(struct ad9545_state *st)
+
+	ad9545_io_update(ref_clk->st);
+
+	regmap_read(ref_clk->st->regmap, AD9545_REFX_STATUS(ref_clk->address), &regval);
```
Hmm, I know this is just debug but I would still do error handling on all API's that might return errors... Ditto for all places where it applies...

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -1402,6 +1420,53 @@ static int ad9545_outputs_setup(struct ad9545_state *st)
+
+	return 0;
+}
+
```
nit: I think in this cases we do not really need the new line so the `#DEFINE` it's just under the fileops

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -1402,6 +1420,53 @@ static int ad9545_outputs_setup(struct ad9545_state *st)
+}
+#else
+#define ad9545_in_clk_debug_init NULL
+#endif
```
also, maybe we could just move all the debugfs stuff to the beginning of the driver and just have one `#ifdef block`? Alternatively and even more neater, if you see that the debugfs interface might increase in the future, we can just move all of this stuff to a ad9545-debugfs file that is only compiled with DEBUGS is set... Hence we can keep the source files clean without any `#ifdefery...`

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -2234,8 +2345,51 @@ static unsigned long ad9545_aux_dpll_clk_recalc_rate(struct clk_hw *hw, unsigned
+	if (regval & AD9545_AUX_DPLL_REF_FAULT)
+		seq_puts(s, "Faulted\n");
+	else
+		seq_puts(s, "Valid\n");
```
nit: I see that in some parts you are just using ternary operator and have this kind of handling with `seq_printf()`. Not really a big deal but I would like to see some consistency. Even though I'm not a fan of using ternary operator, I think in this case it would not hurt that much code readability while reducing considerably the lines of code...

nunojsa reviewed (changes requested):
Overall it looks good. The only thing I would like to see is proper error handling... The other nits; I will leave it up to you

tachicialex commented:
Changelog v1 -> v2:
- Moved all debugfs functions under a single #ifdef
- switched from if else to ternary operator in said places
- added error checks inside debugfs fops functions

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -948,6 +966,130 @@ static int ad9545_io_update(struct ad9545_state *st)
+
+	seq_printf(s, "PLL%d:\n", pll->address);
+
+	regmap_read(pll->st->regmap, AD9545_PLLX_STATUS(pll->address), &regval);
```
Missed the error check here...

nunojsa replied:
Hmm, you need to do `ret = regmap_read(...)`... The same in the other missed place

tachicialex replied:
My bad, fixed

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -948,6 +966,130 @@ static int ad9545_io_update(struct ad9545_state *st)
+	regmap_read(pll->st->regmap, AD9545_PLLX_STATUS(pll->address), &regval);
+	seq_printf(s, "PLL status: %s\n", (regval & AD9545_PLL_LOCKED) ? "Locked" : "Unlocked");
+
+	regmap_read(pll->st->regmap, AD9545_PLLX_OPERATION(pll->address), &regval);
```
ditto...

nunojsa reviewed (changes requested):
Just some missed checks... Also make sure to rebase your branch on top of master so that the checkpatch jobs does not get stuck

tachicialex commented:
Changelog v2 -> v3:
- added checks in missed places
